Polyline_simplification_2/include/CGAL/Polyline_simplification_2/simplify.h

```diff
@@ -154,11 +155,11 @@ class Polyline_simplification_2
       (*it)->set_removable(false);
       ++it;
       for(; it != ite; ++it){
-        if((std::next(it) != ite) && (std::prev(it)== std::next(it))){
+        if((boost::next(it) != ite) && (boost::prior(it)==boost::prior(it))){
```

Comment:
Any idea why the behavior is different?

Reply:
The iterator implementation is most probably buggy (some platforms complains that it is not bidirectional and there are infinite loops on other platforms).